docs/source/scheduling.rst

```diff
@@ -0,0 +1,194 @@
+However if your computation is dominated by processing pure Python objects
+like strings, dicts, or lists,
+then you may want to try one of the process-based schedulers below
+(we currently recommend the distributed scheduler on a local machine).
+
+
+Local Processes
+---------------
+
+*Note: the distributed scheduler described a couple sections below is often a better choice today.
+we encourage readers to continue reading after this section*
+
+.. code-block:: python
+
+   import dask.multiprocessing
+   dask.set_options(get=dask.multiprocessing.get)  # overwrite default with multiprocessing scheduler
+
+
+The multiprocessing scheduler executes computations with a local ``multiprocessing.Pool``.
+It is lightweight to use and requires no setup.
+Every task and all of its dependencies are shipped to a local process,
+executed, and then their result is shipped back to the main process.
+This means that it is able to bypass issues with the GIL and provide parallelism
+even on computations are are dominated by Pure Python code,
+such as those that process strings, dicts, and lists.
+
+However moving data to remote processes and back can introduce performance penalties,
+particularly when the data being transferred between processes is large.
+The multiprocessing scheduler is an excellent choice when workflows are relatively linear,
+and so do not involve significant inter-task data transfer,
+and when inputs and outputs are both small, like filenames and counts.
+
+This is common in basic data ingestion workloads,
+such as those are common in :doc:`dask bag <bag>`,
+where the multiprocessing scheduler is the default.
+
+.. code-block:: python
+
+   >>> import dask.bag as db
+   >>> db.read_text('*.json').map(json.loads).pluck('name').frequencies().compute()
+   {'alice': 100, 'bob': 200, 'charlie': 300}
+
+For more complex workloads,
+where large intermediate results may be depended upon by multiple downstream tasks,
+we generally recommend the use of the distributed scheduler on a local machine.
+The distributed scheduler is more intelligent about moving around large intermediate results.
+
+
+Single Thread
+-------------
+
+.. code-block:: python
+
+   import dask
+   dask.set_options(get=dask.local.get_sync)  # overwrite default with single-threaded scheduler
+
+The single-threaded synchronous scheduler executes all computations in the local thread,
+with no parallelism at all.
+This is particularly valuable for debugging and profiling,
+which are more difficult when using threads or processes.
+
+For example, when using IPython or Jupyter notebooks the ``%debug``, ``%pdb``, or ``%prun`` magics
+will not work well when using the parallel dask schedulers;
+they were not designed to be used in a parallel computing context.
+However if you run into an exception and want to step into the debugger
+you may wish to rerun your computation under the single-threaded scheduler,
+where these tools will function properly.
+
+
+Dask.Distributed (local)
```

Comment:
Using a module path with title case seems wrong to me. Either `dask.distributed (local)` or `Dask Distributed (local)` sound better. Same for below.

Reply:
Agreed.  Thanks for the suggestion.  I think I've hit all of the various links.

Reply:
Done